### ISSUE TITLE
OAuth login with Webauthn not working in Webview

### ISSUE THREAD

**Issue by o-** [RocketChat/Rocket.Chat.ReactNative#5681]
### Description:

We set up Rocket with external OAuth using Keycloak. When Keycloak is configured for second factor authentication using Webauthn, authentication fails with the Rocket native Android app.

### Environment Information:

- Rocket.Chat Server Version: 6.8.0
- Rocket.Chat App Version: 4.48.0 and 4.49.0
- Device Name:  Pixel 6a
- OS Version: Android 14

### Steps to reproduce:

1. Setup a keycloak instance, see e.g. https://www.keycloak.org/getting-started/getting-started-docker
2. Setup rocket chat with keycloak, see https://docs.rocket.chat/use-rocket.chat/authentication/saml/keycloak
3. Configure webauthn. see e.g. https://keycloak.ch/keycloak-tutorials/tutorial-webauthn/
4. On the phone visit https://<keycloak-url>/realms/<realm-name>/account/#/security/signingin and click "Set up security Key" -> Register and add the phone as a second factor.
5. Open Rocket Android app and enter the workspace name.
6. Click on "Continue with Keycloak Login"
7. Provide credentials to Keycloak
8. Try WebAuthn Authenticator as second factor

### Expected behavior:

WebAuthn should work. When I log in with Chrome on the same phone I am asked for the screen pin and the phone successfully authenticates using WebAuthn with Keycloak.

### Actual behavior:

Response "Failed to authenticate by the Security Key" from Keycloak.

Probably the reason is that Android Webview does not support webauthn, see https://groups.google.com/a/chromium.org/g/blink-dev/c/qCJhuuZH5p0 .

### Additional context:

The best would be to open external OAuth authentification links using the device browser instead of opening them inside the app. Most other apps when configured with OIC or OAuth follow this approach.

**Comment by diegolmello** [comment]
Hey. We have plans to make it possible to make login on external browsers which would add support to physical security keys to all services (we did it to Google already https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/2703 https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/2284), but we can't do it right now, sadly. There are more important items atm.

**Comment by JASIM0021** [comment]
working on it

**Comment by KramNamez** [comment]
That's a shame. Rocket.Chat is the last service we have that doesn't work nicely with Passkeys, and the only reason we still have to support TOTPs for MFA, so we'd love to see that change at some point.

Do you have _any_ idea when you'll get around to it? Any sort of roadmap, or is it purely "we'll get around to it eventually, hopefully"? (It's understandable if it's the latter, though obviously I'm hoping for something more concrete.)

**Comment by milton-rucks** [comment]
Hey @KramNamez 

Unfortunately, the answer is that we'll get around to it eventually.

We are currently focused on closing accessibility gaps and new end-to-end encryption requirements on mobile.

**Comment by KramNamez** [comment]
Fair enough - those are both pretty obviously huge and important topics too. Fingers crossed that you get around to it shortly after those :)

**Comment by Czujackt** [comment]
Any Update on this?

**Comment by aenonGit** [comment]
Any news here?

**Comment by machadolucas** [comment]
This is also an issue for me and it would be nice to have any updates on this.
We use AWS Cognito as OAuth provider with passkeys and our users are always confused that it does not work.

**Comment by steinwelberg** [comment]
First of all, thanks for the great product! 

Passkeys are really becoming the standard for strong authentication nowadays, not being able to support passkeys on iOS is becoming a real problem TBH. Is there any timeline on fixing this? 🙏

**Comment by machadolucas** [comment]
I've solved this on our own fork of the repository for our whitelabel app.
The issue as pointed above is that in-app browsers in any platform (used for OAuth authentication) do not support passkeys. The solution was copying the mechanism that works for Google authentication: Instead of opening the OAuth login with in-app browser, it opens on the device native browser, and after authentication, it uses deep linking to redirect back to the app with the token.
I've used AI to make that implementation and it works.

**Comment by diegolmello** [comment]
@machadolucas That's great! Can you contribute back to the community? I'd be happy to review your code 🙏🏼 

@steinwelberg I don't have a timeline, but this is something we've been discussing and we plan to deliver a first step soon.
The idea will be:
- add `Login with browser` button on mobile apps
- button will open the workspace on an external browser
  - no webview, so all permissions are there
  - it'll open the ws on a new state specifically to handle such login
- since it's the same experience like in web, all login services will be available
- the user makes the login and the workspace deep links the user back to the mobile app with the credentials

Let me know if this feature would fix the issues you're seeing. Thanks!

**Comment by steinwelberg** [comment]
That would be a nice first step to at least prevent the dead-end we are seeing now.

**Comment by Czujackt** [comment]
Hi,
i have checked the new Feature but it doesn't work. When I click on the Button for SAML Login the Browser Window is opened and no passkey is possible. So what do I wrong?
